src/main/java/uk/gov/pay/adminusers/resources/UserResource.java

```diff
@@ -79,6 +79,29 @@ public Response getUser(@PathParam("externalId") String externalId) {
                 .orElseGet(() -> Response.status(NOT_FOUND).build());
     }
 
+    @Path(USER_PROFILES_RESOURCE)
+    @GET
+    @Produces(APPLICATION_JSON)
+    @Consumes(APPLICATION_JSON)
+    public Response getUserProfiles(JsonNode payload) {
+        if (payload != null && payload.get(GetUserProfilesRequest.FIELD_EXTERNAL_IDS) != null) {
+            logger.info("Get UserProfiles request - [ {} ]", payload.toString());
+
+            List<String> externalIds = newArrayList(payload.get(GetUserProfilesRequest.FIELD_EXTERNAL_IDS).elements())
+                    .stream().map(externalIdNode -> externalIdNode.textValue()).collect(Collectors.toList());
+
+            List<UserProfile> userProfiles = userServices.findUserByExternalIds(externalIds);
+
+            if (!userProfiles.isEmpty()) {
+                return Response.status(OK).type(APPLICATION_JSON).entity(Collections.unmodifiableMap(Stream
```

Comment:
I would leave all the logic to the UserServices. This resource would parse the externalIds and response back whatever the service returns. I would not return a not found at all because a GET is semantic through its Request Uri is malformed this should return always a 200 (when the payload is correct) with empty list if no users found or populated otherwise. 400 for invalid payload. 

How many ids are going to be requested? why not a get request with a list of ids rather than a body?